### PR TITLE
Make 'make check' more consistent across projects

### PR DESCRIPTION
**Pull request by bcrocker15**
Makefile:
Add 'build-test-image' target, (possibly) distinct from
'build' target.

README.md:
Add mention of 'make build-test-image'

Add CONTRIBUTING.md stub with pointer back to README.md.

Signed-off-by: Ben Crocker <bcrocker@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -36,6 +37,10 @@ clean:
 build:
 	$(CONTAINER_ENGINE) build -t $(IMAGE_NAME) -f Containerfile --network host .
 
+build-test-image:
+	@# In case TEST_IMAGE and IMAGE_NAME are not the same:
+	$(CONTAINER_ENGINE) build -t $(TEST_IMAGE) -f Containerfile --network host .
+
```
Comment (csomh) [inline Makefile:43]: Probably the right thing to do here is to separate break out test dependencies and install them in `TEST_IMAGE` which is different than `IMAGE_NAME` (similar to `tokman`, for example), as running the service shouldn't required pytest and flexmock installed.

Comment (bcrocker15) [reply Makefile:43]: I created a Containerfile.tests =
Containerfile + install python3-pytest python3-flexmock

and ran

%%% make build-test-image
This appeared to go just fine; in particular, it seemed to
install python3-pytest just fine.
...
STEP 7: RUN $package_manager install epel-release     && $package_manager install ansible python3-pip python3-pytest python3-flexmock dnf-plugins-core     && ansible-playbook -vv -c local -i localhost, /src/install-deps.yml     && ansible-playbook -vv -c local -i localhost, /src/install-deps-worker.yml     && $package_manager clean all
...
Installing:
 ansible                    noarch  2.9.21-1.el8                        epel        17 M
 dnf-plugins-core           noarch  4.0.18-4.el8                        baseos      69 k
 python3-flexmock           noarch  0.10.4-10.el8                       epel        57 k
 python3-pip                noarch  9.0.3-19.el8                        appstream   20 k
 python3-pytest             noarch  3.4.2-11.el8                        appstream  1.3 M
...

%%% make check-in-container
This ended up in

/usr/bin/podman run \
        -ti --rm \
        -v /home/packit-all/dist-git-to-source-git/dist2src:/usr/local/lib/python3.6/site-packages/dist2src:Z \
        -v /home/packit-all/dist-git-to-source-git/packitpatch:/usr/bin/packitpatch:Z \
        -v /home/packit-all/dist-git-to-source-git/macros.packit:/usr/lib/rpm/macros.d/macros.packit:Z \
        -v /home/packit-all/dist-git-to-source-git/tests:/tests:Z \
        -v /home/packit-all/dist-git-to-source-git:/src:Z \
        -u 1000 \
        -w / \
         \
        quay.io/packit/dist2src:dev pytest --color=yes --showlocals -vv ./tests
Error: executable file `pytest` not found in $PATH: No such file or directory: OCI not found
make: *** [Makefile:63: check-in-container] Error 127

i.e., just the opposite of what one would expect!

Comment (csomh) [reply Makefile:43]: Without `Containerfile.tests` is hard to tell what was happening above.

But, similar to the comments in `tokman`: Containerfile.tests should start with `FROM quay.io/packit/dist2src:dev` and then `dnf install` the test dependencies.

Comment (bcrocker15) [reply Makefile:43]: OK, so I think you're saying:

ARG base_image=quay.io/centos/centos:8
ARG package_manager
FROM $base_image

in Containerfile should become

ARG base_image=quay.io/packit/dist2src:dev
ARG package_manager
FROM $base_image

in Containerfile.tests

**Review by csomh — APPROVED** [review]

**Comment by csomh** [comment]
@bcrocker15 I've made a few changes on this before marking it to be merged.

**Pull request merged**